### ISSUE TITLE
H264 streaming in Flash Player throws error, but only on the watch page

### ISSUE THREAD

**Issue by TheBigFatOof** [ftde0/yt2009#139]
The default Flash Player works fine in FLV mode, but when switching to H264 mode, I get "An error occurred, please try again later," on any video I try.

However, if I load the player directly through "/watch.swf?video_id=[whatever]/mp4", then it streams in H264 MP4 mode just fine (except with no option for HQ/HD). So it's not that the Flash Player's _incapable_ of streaming in H264 mode; it just won't do it on the normal watch page.

Flash version is 32.0.0.371, running under Windows XP SP3. I've tried these browsers: Firefox 3.6.28, Firefox 52.9.0 ESR, Chrome 49.0.2623.112, and New Moon 27.10.0. All give the same results.

**EDIT:** Tried much earlier Flash version from July 2009 (10.0.32.18). Nothing at all changed.

**Comment by ftde0** [comment]
try using the compatibility fixes available at /toggle_f.

**Comment by TheBigFatOof** [comment]
Thank you, how'd I overlook that? I must've been in /toggle_f a dozen times, with that staring me in the face, and my stupid brain just screened it out.

It seems that toggling "no quality list" fixes the error. Of course, the quality list is one of the reasons I wanted H264 playback, so that's a little disappointing. Is it known why some systems can use the quality list and some can't?

**Comment by ftde0** [comment]
sorry for the late reply, h264 playback is entirely up to the setup you're using. it might be the OS you're using, it might be your browser or even your flash version.

**Comment by TheBigFatOof** [comment]
> sorry for the late reply, h264 playback is entirely up to the setup you're using. it might be the OS you're using, it might be your browser or even your flash version.

Yeah, but what I mean is, are there specific setups that are KNOWN to cause these incompatibilities (versus ones that reliably work)?

**Comment by ftde0** [comment]
no input, assuming solved? reopen if needed

**Comment by TheBigFatOof** [comment]
> no input, assuming solved? reopen if needed

Sorry for necropost, only just now got around to fixing the problem.

Turned out it was simple: post_config_setup was using 127.0.0.1 as the IP. Changing it to the proper network IP fixed it. Yes, the very first thing YT2009 tells you not to do, I was dumb enough to do, and the very first thing I should've tried to fix this, I didn't.

So, word to the wise: if this happens to you, check that your config is set to the right IP.
